### DISCUSSION TITLE
Controlling execution order of workflows triggered by the same webhook trigger

### DISCUSSION THREAD

**Discussion by lukemmtt** [codemagic-ci-cd/codemagic-docs#2612] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Configure any two workflows with the same webhook trigger (e.g. "**build** on push to `main`")

### Expected results

I would expect there to be some means to control which workflow executes first: either explicitly, or implicitly by way of specifically configuring one trigger before the other.

### Actual results

The order in which the workflows are created or configured does not appear to matter, and there is no explicit control.

Specifically:
- Workflow _creation order_ seems to be the only meaningful influence over the webhook-trigger-initiation order, but it is minimal and ultimately a **race condition**—most of the time my iOS builds start first, but then it's a **toss-up** as to whether my Android workflow will execute before my Mac workflow, or vice versa.

**Comment by dtrdic** [comment]
Hey @lukemmtt, can you please provide a use case for why this would be beneficial to you? If you make changes in main.dart, both workflows will be triggered simultaneously (if you have purchased an additional concurrency). Why do you need to control the order in which the workflows are executed? This would help me to better understand your needs

**Reply by lukemmtt** [reply]
Oh certainly, good question and I certainly should have mentioned: the biggest reason is speed to get to a testable build—the same reason I presume someone might want to pay for additional concurrencies, except I'm just one person testing and releasing, so concurrent executions would be overkill.

In other words, in the context of my workflow, consistent execution order would practically give me the same value as two additional concurrencies, for no additional cost. 

Specifically is iOS is my first priority to test after a webhook build is triggered—thats the platform I use on my personal mobile device. If that version is good, then I will often just wait for the macOS version to also finish, and then finish the release notes for both of them and submit them both to Apple manually. By the time I finish with that, the Android build will hopefully be done processing.

**Reply by dtrdic** [reply, marked as answer]
hey @lukemmtt unfortunately, there's no workaround for this scenario except purchasing additional concurrencies. I've classified this as a feature request, and the team will take a look at it for future improvements.
Thank you for your feedback on this mater

**Reply by lukemmtt** [reply]
Thanks David, I certainly understand—through the course of thinking through this, I can understand that there's not much incentive for Codemagic to "fix" this 😅 I'll manage without. Thanks for logging the request though!
